### PR TITLE
Added error message when trying to add deleted device to job

### PR DESCRIPTION
**Pull request by ct-anaalbic**
Signed-off-by: ana.albic.comtrade <Ana.Albic@comtrade.com>

Brief description of the PR.
Error message has been added when trying to add deleted device to job.

**Related Issue**
This PR fixes/issue #2221 

**Description of the solution adopted**
Added error message instead of "Not available" text in the clientId column when the device that was deleted is added to the job.

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

**Comment by LeoNerdoG** [comment]
I have verified that this PR successfully resolves issue #2221 from GUI side. Can one of the commiters check the code?

File: service/api/src/main/java/org/eclipse/kapua/KapuaEntityNotFoundException.java [hunk 1]
```diff
@@ -52,6 +52,12 @@ public KapuaEntityNotFoundException(String entityType, KapuaId entityId) {
         this.entityId = entityId;
     }
 
+    public KapuaEntityNotFoundException(String entityType) {
```
Comment (Coduz) [inline service/api/src/main/java/org/eclipse/kapua/KapuaEntityNotFoundException.java]: Please remove this and check comment on [GwtJobTargetServiceImpl.java](https://github.com/eclipse/kapua/pull/2270/files#diff-bf02bba069510b97ad33fa57f2702882)

File: console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/server/GwtJobTargetServiceImpl.java [hunk 2]
```diff
@@ -112,18 +114,23 @@
...
+                         jobTargetCreator.setJobTargetId(GwtKapuaCommonsModelConverter.convertKapuaId(gwtJobTargetCreator.getJobTargetId()));
+
+                         // Create the Job Target
+                         JobTarget jobTarget = JOB_TARGET_SERVICE.create(jobTargetCreator);
+
+                         // convert to GwtJobTarget and return
+                         gwtJobTargetList.add(KapuaGwtJobModelConverter.convertJobTarget(jobTarget));
+                         //
+                    } else {
+                         throw new KapuaEntityNotFoundException(Device.TYPE);
```
Comment (Coduz) [inline console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/server/GwtJobTargetServiceImpl.java]: Please add the JobTargetId as a parameter, so the user has feedback on which device is not found.

File: console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/server/GwtJobTargetServiceImpl.java [hunk 3]
```diff
@@ -112,18 +114,23 @@
...
-                //
-                // Create the Job Target
-                JobTarget jobTarget = JOB_TARGET_SERVICE.create(jobTargetCreator);
-
-                // convert to GwtJobTarget and return
-                gwtJobTargetList.add(KapuaGwtJobModelConverter.convertJobTarget(jobTarget));
+                } 
+                    Device device = DEVICE_REGISTRY_SERVICE.find(KapuaEid.parseCompactId(gwtJobTargetCreator.getScopeId()), KapuaEid.parseCompactId(gwtJobTargetCreator.getJobTargetId()));
+                    if(device != null) {
+                         KapuaId creatorScopeId = KapuaEid.parseCompactId(gwtJobTargetCreator.getScopeId());
```
Comment (Coduz) [inline console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/server/GwtJobTargetServiceImpl.java]: Fix indentation. 

If you are using Eclipse IDE we have formatting and cleanup profiles under `dev-tools/src/main/eclipse/codeStyle/` 

`

**Comment by LeoNerdoG** [comment]
Hi @Coduz @ct-anaalbic,
I have retested this and I have to say that the ID that is in error message I do not like 😄 
I think the previous version was much better (without the ID part in error message) as this does not carry any specific information for end user. 

**Screenshot:**
screenshot 2018-12-19 at 15 32 10

**Comment by Coduz** [comment]
HI @LeoNerdoG ,

we can trick the thing by adding to `GwtJobTargetCreator` a new field `clientId`. By populating that value, we can retrieve it on the `GwtJobTargetServiceImpl` and use `getClientId()` instead of `getJobTargetId()` when throwing the exception.

Would it work? 

Regards, 

_Alberto_

**Comment by LeoNerdoG** [comment]
@Coduz If it works or not, we will see 😄 
I think this is a good solution if it would work, because everything is better than a "raw" client ID from database. 
Ana is already working on it. 

Regards,
Leo

**Comment by LeoNerdoG** [comment]
@Coduz @ct-anaalbic 
I have checked the latest changes and I can confirm that this is OK from GUI side. 
I am also attaching a screenshot. 

screenshot 2018-12-21 at 09 51 20

**Pull request merged**